exercises/leap/example.scala

```diff
@@ -1,5 +1,6 @@
-case class Year(year: Int) {
-  lazy val isLeap: Boolean = divisibleBy(4) && (divisibleBy(400) || !divisibleBy(100))
+object Year {
+  def isLeap(year: Int): Boolean =
+    divisibleBy(year, 4) && (divisibleBy(year, 400) || !divisibleBy(year, 100))
 
-  private def divisibleBy(i: Int) = year % i == 0
+  private def divisibleBy(year: Int, i: Int) = year % i == 0
```

Comment:
You could consider to have this as a nested function inside `isLeap` so that you can omit the `year` parameter again.